react/features/base/conference/middleware.any.ts

```diff
@@ -195,64 +198,68 @@ function _conferenceFailed({ dispatch, getState }: IStore, next: Function, actio
         break;
     }
     case JitsiConferenceErrors.CONFERENCE_MAX_USERS: {
-        if (typeof APP === 'undefined') {
-            // in case of max users(it can be from a visitor node), let's restore
-            // oldConfig if any as we will be back to the main prosody
-            const newConfig = restoreConferenceOptions(getState);
-
-            if (newConfig) {
-                dispatch(overwriteConfig(newConfig)) // @ts-ignore
-                    .then(dispatch(conferenceWillLeave(conference)))
-                    .then(conference.leave())
-                    .then(dispatch(disconnect()))
-                    .then(dispatch(connect()));
-            }
+        dispatch(showErrorNotification({
+            hideErrorSupportLink: true,
+            descriptionKey: 'dialog.maxUsersLimitReached',
+            titleKey: 'dialog.maxUsersLimitReachedTitle'
+        }, NOTIFICATION_TIMEOUT_TYPE.LONG));
+
+        // In case of max users(it can be from a visitor node), let's restore
+        // oldConfig if any as we will be back to the main prosody.
+        const newConfig = restoreConferenceOptions(getState);
+
+        if (newConfig) {
+            dispatch(overwriteConfig(newConfig)) // @ts-ignore
+                .then(dispatch(conferenceWillLeave(conference)))
+                .then(conference.leave())
+                .then(dispatch(disconnect()))
+                .then(dispatch(connect()));
         }
 
         break;
     }
     case JitsiConferenceErrors.OFFER_ANSWER_FAILED:
         sendAnalytics(createOfferAnswerFailedEvent());
         break;
+
     case JitsiConferenceErrors.REDIRECTED: {
-        // once conference.js is gone this can be removed and both
-        // redirect logics to be merged
-        if (typeof APP === 'undefined') {
-            const newConfig = getVisitorOptions(getState, error.params);
-
-            if (!newConfig) {
-                logger.warn('Not redirected missing params');
-                break;
-            }
+        const newConfig = getVisitorOptions(getState, error.params);
 
-            const [ vnode ] = error.params;
+        if (!newConfig) {
+            logger.warn('Not redirected missing params');
+            break;
+        }
 
-            dispatch(overwriteConfig(newConfig)) // @ts-ignore
-                .then(dispatch(conferenceWillLeave(conference)))
-                .then(conference.leave())
-                .then(dispatch(disconnect()))
-                .then(dispatch(setIAmVisitor(Boolean(vnode))))
+        const [ vnode ] = error.params;
 
-                // we do not clear local tracks on error, so we need to manually clear them
-                .then(dispatch(destroyLocalTracks()))
-                .then(dispatch(connect()));
-        }
+        dispatch(overwriteConfig(newConfig)) // @ts-ignore
+            .then(dispatch(conferenceWillLeave(conference)))
+            .then(conference.leave())
```

Comment:
In the previous version for web we were checking if the room is actually joined before leaving. We don't need this anymore?

Reply:
Yep, this is a cleanup. And I did push a change just now, where we do not mark the REDIRECT error as recoverable so we clean up the ljm state, something the code already does here:
https://github.com/jitsi/jitsi-meet/blob/698fef7c717302d371af50e2197f7ccab48a6a70/react/features/base/conference/middleware.any.ts#L246
So no, that is a cleanup ... in the previous web version it was ljm internally cleaning up ... and now we reuse this ^ 
Maybe we need to move all those cleanups inside ljm itself ... but it is not a new thing juggling with those ...